hudi-integ-test/src/main/java/org/apache/hudi/integ/testsuite/helpers/DFSTestSuitePathSelector.java

```diff
@@ -62,19 +67,26 @@ public DFSTestSuitePathSelector(TypedProperties props, Configuration hadoopConf)
         lastBatchId = 0;
         nextBatchId = 1;
       }
-
-      log.info("Using DFSTestSuitePathSelector, checkpoint: " + lastCheckpointStr + " sourceLimit: " + sourceLimit
-          + " lastBatchId: " + lastBatchId + " nextBatchId: " + nextBatchId);
       // obtain all eligible files for the batch
       List<FileStatus> eligibleFiles = new ArrayList<>();
       FileStatus[] fileStatuses = fs.globStatus(
           new Path(props.getString(Config.ROOT_INPUT_PATH_PROP), "*"));
+      // Say input data is as follow input/1, input/2, input/5 since 3,4 was rolled back and 5 is new generated data
+      // checkpoint from the latest commit metadata will be 2 since 3,4 has been rolled back. We need to set the
+      // next batch id correctly as 5 instead of 3
+      Optional<String> correctBatchIdDueToRollback = Arrays.stream(fileStatuses)
+          .map(f -> f.getPath().toString().split("/")[f.getPath().toString().split("/").length - 1])
+          .min((bid1, bid2) -> Integer.min(Integer.parseInt(bid1), Integer.parseInt(bid2)));
```

Comment:
sorry, are we not looking for max value here. why min ? 
for eg, input/1, input/2, input/5. in this, we are interested in "5" right?

Reply:
Had some uncomitted changes, pushed it now